pkg/controller/certificates/util.go

```diff
@@ -167,6 +168,9 @@ func RequestMatchesSpec(req *cmapi.CertificateRequest, spec cmapi.CertificateSpe
 	if !util.EqualKeyUsagesUnsorted(req.Spec.Usages, spec.Usages) {
 		violations = append(violations, "spec.usages")
 	}
+	if !ExtraNameFieldsEqual(x509req.Subject.Names, spec.Subject.ExtraNames) {
```

Comment:
Could this be a DeepEquals?

Reply:
Unfortunately, no. There's some slightly more complex logic here for equals.